src/toil/jobStores/aws/jobStore.py

```diff
@@ -149,22 +151,44 @@ class AWSJobStore(AbstractJobStore):
     partitioned into chunks of 1024 bytes and each chunk is stored as a an attribute of the SDB
     item representing the job. UUIDs are used to identify jobs and files.
     """
+    class Locator(namedtuple('Locator', ('region', 'namePrefix')), AbstractJobStore.Locator):
+        """
+        Parses a job store locator string into an AWS locator object with the following attributes:
+            * region: The AWS region of the job store as a string.
+            * namePrefix: The name prefix of the job store as a string.
 
-    @classmethod
-    def loadOrCreateJobStore(cls, locator, config=None, **kwargs):
-        region, namePrefix = locator.split(':')
-        if not cls.bucketNameRe.match(namePrefix):
-            raise ValueError("Invalid name prefix '%s'. Name prefixes must contain only digits, "
-                             "hyphens or lower-case letters and must not start or end in a "
-                             "hyphen." % namePrefix)
-        # Reserve 13 for separator and suffix
-        if len(namePrefix) > cls.maxBucketNameLen - cls.maxNameLen - len(cls.nameSeparator):
-            raise ValueError("Invalid name prefix '%s'. Name prefixes may not be longer than 50 "
-                             "characters." % namePrefix)
-        if '--' in namePrefix:
-            raise ValueError("Invalid name prefix '%s'. Name prefixes may not contain "
-                             "%s." % (namePrefix, cls.nameSeparator))
-        return cls(region, namePrefix, config=config, **kwargs)
+        The syntax for an AWS job store locator string is as follows:
+            aws:<AWS region>:<name prefix>
+
+        For name prefix syntax see exception messages in parse method below.
+        """
+        @property
+        def jobStore(self):
+            return AWSJobStore
+
+        @classmethod
+        def parse(cls, l):
+            assert not isinstance(l, str)
+            if len(l) != 2:
+                raise ValueError("The job store locator '%s' is invalid." % ':'.join(l))
+            elif not AWSJobStore.bucketNameRe.match(l[1]):
+                raise ValueError("Invalid prefix '%s'. Name prefixes must contain only digits, "
+                                 "hyphens or lower-case letters and must not start or end in a "
+                                 "hyphen." % l[1])
+            elif (len(l[1]) >
+                  AWSJobStore.maxBucketNameLen -
```

Comment:
This sum is very ugly. If moving the constants to Locator does not require wrapping in those to methods you mention, but allows you to eliminate the ugliness here, I would prefer moving them. Otherwise leave as is, but operators should be at the beginning of continuations, not the end of continued lines.